src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -3299,6 +3347,54 @@ export class StudyViewPageStore {
         return this.customChartsPromises[uniqueKey];
     }
 
+    public getCustomChartDataBinCount(chartMeta: ChartMeta) {
+        const uniqueKey: string = chartMeta.uniqueKey;
+
+        if (!this.clinicalDataBinPromises.hasOwnProperty(uniqueKey)) {
+            this.clinicalDataBinPromises[uniqueKey] = remoteData<DataBin[]>({
+                await: () => {
+                    return [this.selectedSamples];
+                },
+                invoke: () => {
+                    let bin : DataBin[] = [];
+                        const cases = this._customChartsSelectedCases.get(uniqueKey);
+
+                        if (cases) {
+                            const selectedSapmleIds = _.map(this.selectedSamples.result, (sample: Sample) => sample.sampleId);
+                            const filteredCases = cases.filter((caseBeforeFilter: CustomChartIdentifierWithValue) => selectedSapmleIds.includes(caseBeforeFilter.sampleId));
+                            const matchedSampleIds = _.map(filteredCases, (selectedCase: CustomChartIdentifierWithValue) => selectedCase.sampleId);
+                            const casesGroup = _.groupBy(filteredCases, 'value');
```

Comment:
I haven't changed this one since the _.keyBy will pick the last occurrence of the key in a collection. So I cannot get the count of the items in each group.